src/main/java/com/manywho/services/dummy/account/repositories/AccountRepository.java

```diff
@@ -18,44 +20,104 @@ public AccountRepository(Sql2o sql2o) {
 
     public Account find(String id) {
         String sql = "SELECT * FROM account WHERE id = :id";
-
         try (Connection connection = sql2o.open()) {
             return connection.createQuery(sql)
                     .addParameter("id", id)
                     .executeAndFetchFirst(Account.class);
         }
     }
 
-    public List<Account> findAll(ListFilter filter) {
-        String sql = "SELECT * FROM account";
-
+    public Account findAll(String filterBy, String value) {
+        String sql = "SELECT * FROM account WHERE " + filterBy + "= :" + value;
         try (Connection connection = sql2o.open()) {
+            return connection.createQuery(sql)
+                    .addParameter(filterBy, filterBy)
+                    .executeAndFetchFirst(Account.class);
+        }
+    }
+
+    public List<Account> findAll(ListFilter filter) throws ServiceProblemException {
+
+        String sql = "";
+        if (filter != null && filter.getWhere() != null && filter.getWhere().size() != 0) {
+            String columnname = filter.getWhere().get(0).getColumnName();
```

Comment:
could you change to camel case here, this code only works when filtering with one value I think it could be easy to allow filter by many values. But if you want to keep it very simple you should throw an error if the size is > 1

Reply:
I've added another if condition to throw an error when size isn't 1.